docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,757 @@
+            (0a:58:64:41:00:02)"]
+        end
+        VM -->layer2-switch
+    end
+     subgraph node3
+        GR-node3["GR-node3 (0a:58:64:41:00:04)"]
+    end
+    GR-node1 <-->|"remote 100.65.0.2  (0a:58:64:41:00:02)"| layer2-switch
+    GR-node2 <-->|"router 203.203.0.1   (0a:58:64:41:00:03)"| layer2-switch
+    GR-node3 <-->|"remote 100.65.0.4   (0a:58:64:41:00:04)"| layer2-switch
+
+    class VM vmStyle;
+```
+
+ipv6:
+```mermaid
+flowchart TD
+    classDef vmStyle fill:#0000FF,stroke:#ADD8E6,stroke-width:2px;
+
+    subgraph node1["node1 (DOWN)"]
+
+        GR-node1["GR-node1 (fe80::858:64ff:fe41:2)"]
+    end
+    subgraph node2
+        GR-node2["GR-node2 (fe80::858:64ff:fe41:3)"]
+        layer2-switch["Layer2 Switch"]
+        subgraph VM["Virtual Machine"]
+
+            class VM vmStyle;
+            route["default gw
+            down(fe80::858:64ff:fe41:2)
+            fe80::858:64ff:fe41:3
+            fe80::858:64ff:fe41:4"]
+        end
+        layer2-switch--> VM
+    end
+    subgraph node3
+        GR-node3["GR-node3 (fe80::858:64ff:fe41:4)"]
+    end
+
+    GR-node2 --> layer2-switch
+    GR-node3 --> layer2-switch
+
+    class VM vmStyle;
+```
+
+### Layer2 topology limitations for EIP
+
+Also the layer2 current topology has some limitations, when there are multiple
+IPs in an EIP, and if a pod is local to the egress node,
+then we use the external default gateway as the next hop.
+This is not good for GCP or disconnected envs.
+
+## User-Stories/Use-Cases
+
+### Story 1: seamless live migration
+
+As a kubevirt user, I want to live migrate a virtual machine using layer2 primary UDN, 
+so that tcp connections are not broken and downtime is minimum with network configuration 
+not being changed within the virtual machine.
+
+For example: User has a virtual machine serving a video conference using TCP connection and the node
+where is running need to be shut down, so user do a live migration to move to other nodes, the video
+should continue with minimum downtime without changing virtual machine network configuration.
+
+### Story 2: EIP for layer2 limitations
+
+**TODO**
+
+## Proposed Solution
+
+Ther OVN team did introduce a new network topology element [**transit router**](https://www.ovn.org/support/dist-docs/ovn-nb.5.html) that allow to have logical router share
+between OVN zones, this make possible to use a cluster router similar to layer3 topology ovn_cluster_router for layer2
+so the logical router port that is connected to the layer2 switch will have just the .1 address and mac and ipv6 lla generated 
+with it.
+
+There is a [PoC PR](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5067) with passing CI implementing the proposed solution
+
+### Ports, switches and routers topology
+
+This is a big picture of the topology with the transit subnet used to connect the ovn_cluster_router and gateway routers directly without a switch so we have a LRP to LRP connected
```

Comment:
s/a big picture/an overview/

Reply:
I'd rephrase the last part as "directly (without a new logical switch - direct router port connections using the `NB.Logical_Router_Port.peer field` between the GR and the new ovn_cluster_router)."